schutzbot/Jenkinsfile

```diff
@@ -344,6 +344,141 @@ pipeline {
 
 }
 
+pipeline {
+    // test against RHEL nightly builds
+    agent none
+
+    // TODO: do we need to disable triggering via GitHub pull requests ???
+    triggers {
+        cron('@daily')
+    }
+
+    environment {
+        AWS_REGION = "us-east-2"
+        AWS_BUCKET = "imagebuilder-jenkins-testing-use2"
+    }
+
+    options {
+        timestamps()
+        ansiColor('xterm')
+        // Cancel the pipeline if it runs for more than three hours.
+        timeout(
+            time: 3,
+            unit: "HOURS"
+        )
+    }
+    stages {
+
+        stage("Prepare 🤔") {
+            agent { label "schutzbot" }
+            options {
+                // Don't checkout the git repository here. It just clogs
+                // up the Jenkins disk space and does nothing for us.
+                skipDefaultCheckout()
+            }
+            steps {
+                sh (
+                    label: "Get environment variables",
+                    script: "env | sort"
+                )
+                // don't build, use RPMs from the nightly trees
+                sh "schutzbot/prepare-rhel-nightly.sh"
+                stash (
+                    includes: 'osbuild-mock.repo',
+                    name: 'rhel8nightly'
+                )
+                stash (
+                    includes: 'rhel-8.json',
+                    name: 'rhel8json'
+                )
+                stash (
+                    includes: 'COMPOSE_ID',
+                    name: 'compose_id'
+                )
+            }
+        }
+
+        stage("Testing 🍌") {
+            parallel {
+
+                stage('EL8 Base') {
+                    agent { label "rhel8cloudbase && psi && x86_64" }
+                    environment {
+                        TEST_TYPE = "base"
+                    }
+                    steps {
+                        unstash 'rhel8nightly'
+                        unstash 'rhel8json'
+                        run_tests('base')
+                    }
+                    post {
+                        always {
+                            preserve_logs('rhel8-base')
+                        }
+                    }
+                }
+                stage('EL8 Image') {
```

Comment:
Note that we cannot run the image tests against a moving target. So we would need to drop this.

I think the solution for this is to move all the boot-tests from image tests to integration tests, which can be run in any environment.

Reply:
> Note that we cannot run the image tests against a moving target. So we would need to drop this.

To be more precise you mean we can't execute the content verification portion of these tests. OTOH the upload & boot to cloud portion can be executed.

For one there's `-disable-local-boot` and we can very easily add another option, e.g. `-disable-content-check`. OTOH I've been running these same tests internally against snapshots and they were passing so I need to check what exactly is happening in that case.

Reply:
FTR these image tests vs. integration tests are getting kind of duplicate & redundant. We should clean them up. I'll open an issue for that.

Reply:
I agree. We should make the content check and the boot tests completely separate. (And the boot tests should be moved to the integration tests). A bit of rework is required, but I think all the code can easily be reused.

Reply:
QE notes:

- I'm playing around with a way to skip the image info part of these tests but I think it may not be necessary

- ATM we have "EL8 Image" and "EL8.3 Image" stages in the CI pipeline, which differ by the RHN registration command they use, hence the final list of sources. However there is only 1 set of json files for the image tests and inside the manifest section they contain hard-coded URLs for every single RPM. All of this goes directly to osbuild instead of osbuild-composer and I think this is the reason it hasn't failed for me so far in all of my internal executions. 

Essentially the OS version could be a moving target but the content for the image tests is always hardcoded. 

@teg I think the original concern expressed in the first comment is nothing to be worried about ATM and the last 2 commits I've added are not needed. What do you say ?

Reply:
You are absolutely right. Keeping the image tests as-is is harmless. I'm sorry for causing confusion here.

We should still improve on this in the future by splitting all the boot tests from the image tests so the boot tests can easily be run against the latest nightly (or whatever we point them at), rather than the fixed sources. The way it is now we might think we test more than we do.

Reply:
> We should still improve on this in the future by splitting all the boot tests from the image tests so the boot tests can easily be run against the latest nightly (or whatever we point them at), rather than the fixed sources. The way it is now we might think we test more than we do.

Agreed. I've opened https://github.com/osbuild/osbuild-composer/issues/962 so we don't forget and after we're done with the Jenkins configuration my plan is to start working on improving these tests.